pkg/vcs/git_test.go

```diff
@@ -371,3 +373,55 @@ func TestMergeBase(t *testing.T) {
 		t.Fatalf("expected base commit, got %v", mergeCommits)
 	}
 }
+
+func TestGitCustomRefs(t *testing.T) {
+	remoteRepoDir := t.TempDir()
+	remote := MakeTestRepo(t, remoteRepoDir)
+	remote.Git("commit", "--no-edit", "--allow-empty", "-m", "base commit")
+	remote.Git("checkout", "-b", "base_branch")
+	remote.Git("tag", "base_tag")
+
+	// Create a custom ref commit.
+	remote.Git("checkout", "base_branch")
+	remote.Git("checkout", "-b", "temp_branch")
+	remote.Git("commit", "--no-edit", "--allow-empty", "-m", "detached commit")
+	remote.Git("update-ref", "refs/custom/test", "temp_branch")
```

Comment:
I wasn't aware of update-ref before this. Why do you need this anyway? Is the detached commit otherwise garbage collected when temp_branch is deleted or something?

Reply:
May be useful to capture this in a comment.

Reply:
Added a comment.